osu.Game/Screens/OnlinePlay/Matchmaking/Match/BeatmapSelect/SubScreenBeatmapSelect.cs

```diff
@@ -80,6 +82,17 @@ private void onItemDeselected(int userId, long itemId)
             beatmapSelectGrid.SetUserSelection(user, itemId, false);
         }
 
+        private void onSettingsChanged(MultiplayerRoomSettings settings)
+        {
+            if (client.Room!.MatchState is not MatchmakingRoomState matchmakingState)
+                return;
+
+            if (matchmakingState.CandidateItem != -1 || client.Room!.CurrentPlaylistItem.Expired)
```

Comment:
What's the reason for the `Expired` check?

Reply:
I'm actually going to replace this condition with one relying on the stage. I think it's a bit of a much-of-a-muchness since the room settings aren't expected to be changed via anything other than the playlist item changing.

So yeah I can't really explain the guard. Will remove.